cmd/soroban-rpc/internal/db/migrations/02_transactions.sql

```diff
@@ -0,0 +1,16 @@
+-- +migrate Up
+
+-- index to find transactions in ledgers by hash
+CREATE TABLE transactions (
+    hash NCHAR(64) PRIMARY KEY,
+    ledger_sequence INTEGER NOT NULL,
+    application_order INTEGER NOT NULL,
+    -- possibly a useful search index for Soroban-oriented queries, e.g.
+    -- transactions that involve the SAC or have just events in general
+    is_soroban BOOLEAN NOT NULL,
+    FOREIGN KEY (ledger_sequence)
```

Comment:
what's the motivation for using a foreign key constraint here? support for foreign key constraints are disabled in sqlite by default

Reply:
just seemed strictly more correct, but it doesn't really add any value if the code is correct